ensure_str is missing a parameter

I didn't check why your CI does not fail, but trying to package fakeredis 1.9.3 and running pytest to check package integrity fails with this error:

<details>
<summary>[   49s] E   TypeError: ensure_str() takes 2 positional arguments but 4 were given</summary>

```
[   49s] __________________ test_eval_global_variable[FakeStrictRedis] __________________
[   49s] 
[   49s] r = FakeStrictRedis<ConnectionPool<FakeConnection<server=<fakeredis._server.FakeServer object at 0x7f3a26ace250>,db=0>>>
[   49s] 
[   49s]     def test_eval_global_variable(r):
[   49s]         # Redis doesn't allow script to define global variables
[   49s]         with pytest.raises(ResponseError):
[   49s] >           r.eval('a=10', 0)
[   49s] 
[   49s] test/test_lua.py:214: 
[   49s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   49s] /usr/lib/python3.8/site-packages/redis/commands/core.py:5054: in eval
[   49s]     return self._eval("EVAL", script, numkeys, *keys_and_args)
[   49s] /usr/lib/python3.8/site-packages/redis/commands/core.py:5039: in _eval
[   49s]     return self.execute_command(command, script, numkeys, *keys_and_args)
[   49s] /usr/lib/python3.8/site-packages/redis/client.py:1235: in execute_command
[   49s]     return conn.retry.call_with_retry(
[   49s] /usr/lib/python3.8/site-packages/redis/retry.py:46: in call_with_retry
[   49s]     return do()
[   49s] /usr/lib/python3.8/site-packages/redis/client.py:1236: in <lambda>
[   49s]     lambda: self._send_command_parse_response(
[   49s] /usr/lib/python3.8/site-packages/redis/client.py:1211: in _send_command_parse_response
[   49s]     conn.send_command(*args)
[   49s] /usr/lib/python3.8/site-packages/redis/connection.py:797: in send_command
[   49s]     self.send_packed_command(
[   49s] /usr/lib/python3.8/site-packages/redis/connection.py:779: in send_packed_command
[   49s]     self._sock.sendall(item)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_basefakesocket.py:208: in sendall
[   49s]     self._parser.send(data)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_basefakesocket.py:118: in _parse_commands
[   49s]     self._process_command(fields)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_basefakesocket.py:238: in _process_command
[   49s]     result = self._run_command(func, sig, fields[1:], False)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_basefakesocket.py:139: in _run_command
[   49s]     result = func(*args)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_fakesocket.py:1606: in eval
[   49s]     self._check_for_lua_globals(lua_runtime, expected_globals)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_fakesocket.py:1528: in _check_for_lua_globals
[   49s]     unexpected = [self.ensure_str(var, 'utf-8', 'replace')
[   49s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   49s] 
[   49s] .0 = <set_iterator object at 0x7f3a2715c0c0>
[   49s] 
[   49s] >   unexpected = [self.ensure_str(var, 'utf-8', 'replace')
[   49s]                   for var in actual_globals - expected_globals]
[   49s] E   TypeError: ensure_str() takes 2 positional arguments but 4 were given
[   49s] 
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_fakesocket.py:1528: TypeError
[   49s] _______________ test_eval_global_and_return_ok[FakeStrictRedis] ________________
[   49s] 
[   49s] r = FakeStrictRedis<ConnectionPool<FakeConnection<server=<fakeredis._server.FakeServer object at 0x7f3a26b20580>,db=0>>>
[   49s] 
[   49s]     def test_eval_global_and_return_ok(r):
[   49s]         # Redis doesn't allow script to define global variables
[   49s]         with pytest.raises(ResponseError):
[   49s] >           r.eval(
[   49s]                 '''
[   49s]                 a=10
[   49s]                 return redis.status_reply("Everything is awesome")
[   49s]                 ''',
[   49s]                 0
[   49s]             )
[   49s] 
[   49s] test/test_lua.py:220: 
[   49s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   49s] /usr/lib/python3.8/site-packages/redis/commands/core.py:5054: in eval
[   49s]     return self._eval("EVAL", script, numkeys, *keys_and_args)
[   49s] /usr/lib/python3.8/site-packages/redis/commands/core.py:5039: in _eval
[   49s]     return self.execute_command(command, script, numkeys, *keys_and_args)
[   49s] /usr/lib/python3.8/site-packages/redis/client.py:1235: in execute_command
[   49s]     return conn.retry.call_with_retry(
[   49s] /usr/lib/python3.8/site-packages/redis/retry.py:46: in call_with_retry
[   49s]     return do()
[   49s] /usr/lib/python3.8/site-packages/redis/client.py:1236: in <lambda>
[   49s]     lambda: self._send_command_parse_response(
[   49s] /usr/lib/python3.8/site-packages/redis/client.py:1211: in _send_command_parse_response
[   49s]     conn.send_command(*args)
[   49s] /usr/lib/python3.8/site-packages/redis/connection.py:797: in send_command
[   49s]     self.send_packed_command(
[   49s] /usr/lib/python3.8/site-packages/redis/connection.py:779: in send_packed_command
[   49s]     self._sock.sendall(item)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_basefakesocket.py:208: in sendall
[   49s]     self._parser.send(data)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_basefakesocket.py:118: in _parse_commands
[   49s]     self._process_command(fields)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_basefakesocket.py:238: in _process_command
[   49s]     result = self._run_command(func, sig, fields[1:], False)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_basefakesocket.py:139: in _run_command
[   49s]     result = func(*args)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_fakesocket.py:1606: in eval
[   49s]     self._check_for_lua_globals(lua_runtime, expected_globals)
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_fakesocket.py:1528: in _check_for_lua_globals
[   49s]     unexpected = [self.ensure_str(var, 'utf-8', 'replace')
[   49s] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[   49s] 
[   49s] .0 = <set_iterator object at 0x7f3a26cbc480>
[   49s] 
[   49s] >   unexpected = [self.ensure_str(var, 'utf-8', 'replace')
[   49s]                   for var in actual_globals - expected_globals]
[   49s] E   TypeError: ensure_str() takes 2 positional arguments but 4 were given
[   49s] 
[   49s] ../../BUILDROOT/python-fakeredis-1.9.3-0.x86_64/usr/lib/python3.8/site-packages/fakeredis/_fakesocket.py:1528: TypeError
[   49s] =============================== warnings summary ===============================
[   49s] test/test_connection.py::TestFakeStrictRedisConnectionErrors::test_hmset[FakeStrictRedis]
[   49s]   /home/abuild/rpmbuild/BUILD/fakeredis-py-1.9.3/test/test_connection.py:258: DeprecationWarning: FakeStrictRedis.hmset() is deprecated. Use FakeStrictRedis.hset() instead.
[   49s]     r.hmset('name', {'key': 1})
[   49s] 
[   49s] test/test_fakeredis6.py::test_hmset_empty_raises_error[FakeStrictRedis]
[   49s]   /home/abuild/rpmbuild/BUILD/fakeredis-py-1.9.3/test/test_fakeredis6.py:1543: DeprecationWarning: FakeStrictRedis.hmset() is deprecated. Use FakeStrictRedis.hset() instead.
[   49s]     r.hmset('foo', {})
[   49s] 
[   49s] test/test_fakeredis6.py::test_hmset[FakeStrictRedis]
[   49s]   /home/abuild/rpmbuild/BUILD/fakeredis-py-1.9.3/test/test_fakeredis6.py:1548: DeprecationWarning: FakeStrictRedis.hmset() is deprecated. Use FakeStrictRedis.hset() instead.
[   49s]     assert r.hmset('foo', {'k2': 'v2', 'k3': 'v3'}) is True
[   49s] 
[   49s] test/test_fakeredis6.py::test_hmset_wrong_type[FakeStrictRedis]
[   49s]   /home/abuild/rpmbuild/BUILD/fakeredis-py-1.9.3/test/test_fakeredis6.py:1554: DeprecationWarning: FakeStrictRedis.hmset() is deprecated. Use FakeStrictRedis.hset() instead.
[   49s]     r.hmset('foo', {'key': 'value'})
[   49s] 
[   49s] -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[   49s] =========================== short test summary info ============================
[   49s] FAILED test/test_lua.py::test_eval_global_variable[FakeStrictRedis] - TypeErr...
[   49s] FAILED test/test_lua.py::test_eval_global_and_return_ok[FakeStrictRedis] - Ty...
[   49s] =========== 2 failed, 652 passed, 632 skipped, 4 warnings in 44.62s ============

```
</details>

Clearly there is somethring wrong in the last commit:

https://github.com/cunla/fakeredis-py/blob/cc8c2e77ccb9117676cb22fee3bdff38a622d6a2/fakeredis/_fakesocket.py#L1520-L1530

--- a/fakeredis/_basefakeluasupport.py
+++ b/fakeredis/_basefakeluasupport.py
@@ -24,7 +24,7 @@ def _check_for_lua_globals(lua_runtime, expected_globals):
 
 
 def _lua_redis_log(lua_runtime, expected_globals, lvl, *args):
-    _check_for_lua_globals(lua_runtime.globals().keys(), expected_globals)
+    _check_for_lua_globals(lua_runtime, expected_globals)
     if len(args) < 1:
         raise SimpleError(msgs.REQUIRES_MORE_ARGS_MSG.format("redis.log()", "two"))
     if lvl not in REDIS_LOG_LEVELS.values():

--- a/fakeredis/_basefakeluasupport.py
+++ b/fakeredis/_basefakeluasupport.py
@@ -1,0 +1,167 @@
+import functools
+import hashlib
+import itertools
+
+from fakeredis._helpers import SimpleError, SimpleString
+from . import _msgs as msgs
+from ._commands import command, Int
+from ._helpers import REDIS_LOG_LEVELS, REDIS_LOG_LEVELS_TO_LOGGING, LOGGER
+
+
+def _ensure_str(s, encoding, replaceerr):
+    if isinstance(s, bytes):
+        res = s.decode(encoding=encoding, errors=replaceerr)
+    else:
+        res = str(s).encode(encoding=encoding, errors=replaceerr)
+    return res
+
+
+def _check_for_lua_globals(lua_runtime, expected_globals):
+    unexpected_globals = set(lua_runtime.globals().keys()) - expected_globals
+    if len(unexpected_globals) > 0:
+        unexpected = [_ensure_str(var, 'utf-8', 'replace') for var in unexpected_globals]
+        raise SimpleError(msgs.GLOBAL_VARIABLE_MSG.format(", ".join(unexpected)))
+
+
+def _lua_redis_log(lua_runtime, expected_globals, lvl, *args):
+    _check_for_lua_globals(lua_runtime.globals().keys(), expected_globals)
+    if len(args) < 1:
+        raise SimpleError(msgs.REQUIRES_MORE_ARGS_MSG.format("redis.log()", "two"))
+    if lvl not in REDIS_LOG_LEVELS.values():
+        raise SimpleError(msgs.LOG_INVALID_DEBUG_LEVEL_MSG)
+    msg = ' '.join([x.decode('utf-8')
+                    if isinstance(x, bytes) else str(x)
+                    for x in args if not isinstance(x, bool)])
+    LOGGER.log(REDIS_LOG_LEVELS_TO_LOGGING[lvl], msg)
+
+
+class BaseFakeLuaSocket:
+
+    # Script commands
+    # script debug and script kill will probably not be supported
+
+    def _convert_redis_arg(self, lua_runtime, value):
+        # Type checks are exact to avoid issues like bool being a subclass of int.
+        if type(value) is bytes:
+            return value
+        elif type(value) in {int, float}:
+            return '{:.17g}'.format(value).encode()
+        else:
+            # TODO: add the context
+            msg = msgs.LUA_COMMAND_ARG_MSG6 if self.version < 7 else msgs.LUA_COMMAND_ARG_MSG
+            raise SimpleError(msg)
+
+    def _convert_redis_result(self, lua_runtime, result):
+        if isinstance(result, (bytes, int)):
+            return result
+        elif isinstance(result, SimpleString):
+            return lua_runtime.table_from({b"ok": result.value})
+        elif result is None:
+            return False
+        elif isinstance(result, list):
+            converted = [
+                self._convert_redis_result(lua_runtime, item)
+                for item in result
+            ]
+            return lua_runtime.table_from(converted)
+        elif isinstance(result, SimpleError):
+            raise result
+        else:
+            raise RuntimeError("Unexpected return type from redis: {}".format(type(result)))
+
+    def _convert_lua_result(self, result, nested=True):
+        from lupa import lua_type
+        if lua_type(result) == 'table':
+            for key in (b'ok', b'err'):
+                if key in result:
+                    msg = self._convert_lua_result(result[key])
+                    if not isinstance(msg, bytes):
+                        raise SimpleError(msgs.LUA_WRONG_NUMBER_ARGS_MSG)
+                    if key == b'ok':
+                        return SimpleString(msg)
+                    elif nested:
+                        return SimpleError(msg.decode('utf-8', 'replace'))
+                    else:
+                        raise SimpleError(msg.decode('utf-8', 'replace'))
+            # Convert Lua tables into lists, starting from index 1, mimicking the behavior of StrictRedis.
+            result_list = []
+            for index in itertools.count(1):
+                if index not in result:
+                    break
+                item = result[index]
+                result_list.append(self._convert_lua_result(item))
+            return result_list
+        elif isinstance(result, str):
+            return result.encode()
+        elif isinstance(result, float):
+            return int(result)
+        elif isinstance(result, bool):
+            return 1 if result else None
+        return result
+
+    def _lua_redis_call(self, lua_runtime, expected_globals, op, *args):
+        # Check if we've set any global variables before making any change.
+        _check_for_lua_globals(lua_runtime, expected_globals)
+        func, func_name = self._name_to_func(op)
+        args = [self._convert_redis_arg(lua_runtime, arg) for arg in args]
+        result = self._run_command(func, func._fakeredis_sig, args, True)
+        return self._convert_redis_result(lua_runtime, result)
+
+    def _lua_redis_pcall(self, lua_runtime, expected_globals, op, *args):
+        try:
+            return self._lua_redis_call(lua_runtime, expected_globals, op, *args)
+        except Exception as ex:
+            return lua_runtime.table_from({b"err": str(ex)})
+
+    @command((bytes, Int), (bytes,), flags='s')
+    def eval(self, script, numkeys, *keys_and_args):
+        from lupa import LuaError, LuaRuntime, as_attrgetter
+
+        if numkeys > len(keys_and_args):
+            raise SimpleError(msgs.TOO_MANY_KEYS_MSG)
+        if numkeys < 0:
+            raise SimpleError(msgs.NEGATIVE_KEYS_MSG)
+        sha1 = hashlib.sha1(script).hexdigest().encode()
+        self._server.script_cache[sha1] = script
+        lua_runtime = LuaRuntime(encoding=None, unpack_returned_tuples=True)
+
+        set_globals = lua_runtime.eval(
+            """
+            function(keys, argv, redis_call, redis_pcall, redis_log, redis_log_levels)
+                redis = {}
+                redis.call = redis_call
+                redis.pcall = redis_pcall
+                redis.log = redis_log
+                for level, pylevel in python.iterex(redis_log_levels.items()) do
+                    redis[level] = pylevel
+                end
+                redis.error_reply = function(msg) return {err=msg} end
+                redis.status_reply = function(msg) return {ok=msg} end
+                KEYS = keys
+                ARGV = argv
+            end
+            """
+        )
+        expected_globals = set()
+        set_globals(
+            lua_runtime.table_from(keys_and_args[:numkeys]),
+            lua_runtime.table_from(keys_and_args[numkeys:]),
+            functools.partial(self._lua_redis_call, lua_runtime, expected_globals),
+            functools.partial(self._lua_redis_pcall, lua_runtime, expected_globals),
+            functools.partial(_lua_redis_log, lua_runtime, expected_globals),
+            as_attrgetter(REDIS_LOG_LEVELS)
+        )
+        expected_globals.update(lua_runtime.globals().keys())
+
+        try:
+            result = lua_runtime.execute(script)
+        except SimpleError as ex:
+            if self.version == 6:
+                raise SimpleError(msgs.SCRIPT_ERROR_MSG.format(sha1.decode(), ex))
+            raise SimpleError(ex.value)
+        except LuaError as ex:
+            raise SimpleError(msgs.SCRIPT_ERROR_MSG.format(sha1.decode(), ex))
+
+        _check_for_lua_globals(lua_runtime, expected_globals)
+
+        return self._convert_lua_result(result, nested=False)

--- a/fakeredis/_fakesocket.py
+++ b/fakeredis/_fakesocket.py
@@ -9,18 +9,18 @@ import time
 import redis
 
 from . import _msgs as msgs
+from ._basefakeluasupport import BaseFakeLuaSocket
 from ._basefakesocket import BaseFakeSocket
 from ._commands import (
     Key, command, DbIndex, Int, CommandItem, BeforeAny, SortFloat, Float, BitOffset, BitValue, Hash,
     StringTest, ScoreTest, Timeout)
 from ._helpers import (
     PONG, OK, MAX_STRING_SIZE, SimpleError, SimpleString, casematch,
-    BGSAVE_STARTED, REDIS_LOG_LEVELS_TO_LOGGING, LOGGER, REDIS_LOG_LEVELS, casenorm, compile_pattern)
-from ._msgs import LUA_COMMAND_ARG_MSG, LUA_COMMAND_ARG_MSG6
+    BGSAVE_STARTED, casenorm, compile_pattern)
 from ._zset import ZSet
 
 
-class FakeSocket(BaseFakeSocket):
+class FakeSocket(BaseFakeSocket, BaseFakeLuaSocket):
     _connection_error_class = redis.ConnectionError
 
     def __init__(self, server):
@@ -1454,158 +1454,6 @@ class FakeSocket(BaseFakeSocket):
         now_s = now_us // 1000000
         now_us %= 1000000
         return [str(now_s).encode(), str(now_us).encode()]
-
-    # Script commands
-    # script debug and script kill will probably not be supported
-
-    def _convert_redis_arg(self, lua_runtime, value):
-        # Type checks are exact to avoid issues like bool being a subclass of int.
-        if type(value) is bytes:
-            return value
-        elif type(value) in {int, float}:
-            return '{:.17g}'.format(value).encode()
-        else:
-            # TODO: add the context
-            msg = LUA_COMMAND_ARG_MSG6 if self.version < 7 else LUA_COMMAND_ARG_MSG
-            raise SimpleError(msg)
-
-    def _convert_redis_result(self, lua_runtime, result):
-        if isinstance(result, (bytes, int)):
-            return result
-        elif isinstance(result, SimpleString):
-            return lua_runtime.table_from({b"ok": result.value})
-        elif result is None:
-            return False
-        elif isinstance(result, list):
-            converted = [
-                self._convert_redis_result(lua_runtime, item)
-                for item in result
-            ]
-            return lua_runtime.table_from(converted)
-        elif isinstance(result, SimpleError):
-            raise result
-        else:
-            raise RuntimeError("Unexpected return type from redis: {}".format(type(result)))
-
-    def _convert_lua_result(self, result, nested=True):
-        from lupa import lua_type
-        if lua_type(result) == 'table':
-            for key in (b'ok', b'err'):
-                if key in result:
-                    msg = self._convert_lua_result(result[key])
-                    if not isinstance(msg, bytes):
-                        raise SimpleError(msgs.LUA_WRONG_NUMBER_ARGS_MSG)
-                    if key == b'ok':
-                        return SimpleString(msg)
-                    elif nested:
-                        return SimpleError(msg.decode('utf-8', 'replace'))
-                    else:
-                        raise SimpleError(msg.decode('utf-8', 'replace'))
-            # Convert Lua tables into lists, starting from index 1, mimicking the behavior of StrictRedis.
-            result_list = []
-            for index in itertools.count(1):
-                if index not in result:
-                    break
-                item = result[index]
-                result_list.append(self._convert_lua_result(item))
-            return result_list
-        elif isinstance(result, str):
-            return result.encode()
-        elif isinstance(result, float):
-            return int(result)
-        elif isinstance(result, bool):
-            return 1 if result else None
-        return result
-
-    def ensure_str(self, s):
-        return (s.decode(encoding='utf-8', errors='replace')
-                if isinstance(s, bytes)
-                else str(s).encode(encoding='utf-8', errors='replace'))
-
-    def _check_for_lua_globals(self, lua_runtime, expected_globals):
-        actual_globals = set(lua_runtime.globals().keys())
-        if actual_globals != expected_globals:
-            unexpected = [self.ensure_str(var, 'utf-8', 'replace')
-                          for var in actual_globals - expected_globals]
-            raise SimpleError(msgs.GLOBAL_VARIABLE_MSG.format(", ".join(unexpected)))
-
-    def _lua_redis_call(self, lua_runtime, expected_globals, op, *args):
-        # Check if we've set any global variables before making any change.
-        self._check_for_lua_globals(lua_runtime, expected_globals)
-        func, func_name = self._name_to_func(op)
-        args = [self._convert_redis_arg(lua_runtime, arg) for arg in args]
-        result = self._run_command(func, func._fakeredis_sig, args, True)
-        return self._convert_redis_result(lua_runtime, result)
-
-    def _lua_redis_pcall(self, lua_runtime, expected_globals, op, *args):
-        try:
-            return self._lua_redis_call(lua_runtime, expected_globals, op, *args)
-        except Exception as ex:
-            return lua_runtime.table_from({b"err": str(ex)})
-
-    def _lua_redis_log(self, lua_runtime, expected_globals, lvl, *args):
-        self._check_for_lua_globals(lua_runtime, expected_globals)
-        if len(args) < 1:
-            raise SimpleError(msgs.REQUIRES_MORE_ARGS_MSG.format("redis.log()", "two"))
-        if lvl not in REDIS_LOG_LEVELS.values():
-            raise SimpleError(msgs.LOG_INVALID_DEBUG_LEVEL_MSG)
-        msg = ' '.join([x.decode('utf-8')
-                        if isinstance(x, bytes) else str(x)
-                        for x in args if not isinstance(x, bool)])
-        LOGGER.log(REDIS_LOG_LEVELS_TO_LOGGING[lvl], msg)
-
-    @command((bytes, Int), (bytes,), flags='s')
-    def eval(self, script, numkeys, *keys_and_args):
-        from lupa import LuaError, LuaRuntime, as_attrgetter
-
-        if numkeys > len(keys_and_args):
-            raise SimpleError(msgs.TOO_MANY_KEYS_MSG)
-        if numkeys < 0:
-            raise SimpleError(msgs.NEGATIVE_KEYS_MSG)
-        sha1 = hashlib.sha1(script).hexdigest().encode()
-        self._server.script_cache[sha1] = script
-        lua_runtime = LuaRuntime(encoding=None, unpack_returned_tuples=True)
-
-        set_globals = lua_runtime.eval(
-            """
-            function(keys, argv, redis_call, redis_pcall, redis_log, redis_log_levels)
-                redis = {}
-                redis.call = redis_call
-                redis.pcall = redis_pcall
-                redis.log = redis_log
-                for level, pylevel in python.iterex(redis_log_levels.items()) do
-                    redis[level] = pylevel
-                end
-                redis.error_reply = function(msg) return {err=msg} end
-                redis.status_reply = function(msg) return {ok=msg} end
-                KEYS = keys
-                ARGV = argv
-            end
-            """
-        )
-        expected_globals = set()
-        set_globals(
-            lua_runtime.table_from(keys_and_args[:numkeys]),
-            lua_runtime.table_from(keys_and_args[numkeys:]),
-            functools.partial(self._lua_redis_call, lua_runtime, expected_globals),
-            functools.partial(self._lua_redis_pcall, lua_runtime, expected_globals),
-            functools.partial(self._lua_redis_log, lua_runtime, expected_globals),
-            as_attrgetter(REDIS_LOG_LEVELS)
-        )
-        expected_globals.update(lua_runtime.globals().keys())
-
-        try:
-            result = lua_runtime.execute(script)
-        except SimpleError as ex:
-            if self.version == 6:
-                raise SimpleError(msgs.SCRIPT_ERROR_MSG.format(sha1.decode(), ex))
-            raise SimpleError(ex.value)
-        except LuaError as ex:
-            raise SimpleError(msgs.SCRIPT_ERROR_MSG.format(sha1.decode(), ex))
-
-        self._check_for_lua_globals(lua_runtime, expected_globals)
-
-        return self._convert_lua_result(result, nested=False)
 
     @command((bytes, Int), (bytes,), flags='s')
     def evalsha(self, sha1, numkeys, *keys_and_args):

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -8,7 +8,7 @@ name = "fakeredis"
 packages = [
     { include = "fakeredis" },
 ]
-version = "1.9.3"
+version = "1.9.4"
 description = "Fake implementation of redis API for testing purposes."
 readme = "README.md"
 keywords = ["redis", "rq", "django-rq", "rq-scheduler"]
@@ -57,7 +57,7 @@ twine = "4.0.1"
 coverage = "^6.3"
 pytest = "^7.1.2"
 pytest-asyncio = "0.19.0"
-pytest-cov = "^3.0.0"
+pytest-cov = "^4.0.0"
 pytest-mock = "^3.7.0"
 flake8 = "^5.0.4"
 

